docs/flutterweathertutorial.md

```diff
@@ -43,476 +67,2514 @@ We'll be focusing on two endpoints:
+            "wind_direction_compass":"SW",
+            "created":"2020-10-26T00:20:01.840132Z",
+            "applicable_date":"2020-10-26",
+            "min_temp":7.9399999999999995,
+            "max_temp":13.239999999999998,
+            "the_temp":12.825,
+            "wind_speed":7.876886316914553,
+            "wind_direction":246.17046093256732,
+            "air_pressure":997.0,
+            "humidity":73,
+            "visibility":11.037727173307882,
+            "predictability":73
+          }]}
+        ''');
+        when(() => httpClient.get(any())).thenAnswer((_) async => response);
+        final actual = await metaWeatherApiClient.getWeather(locationId);
+        expect(
+          actual,
+          isA<Weather>()
+              .having((w) => w.id, 'id', 4907479830888448)
+              .having((w) => w.weatherStateName, 'state', 'Showers')
+              .having((w) => w.weatherStateAbbr, 'abbr', WeatherState.showers)
+              .having((w) => w.windDirectionCompass, 'wind',
+                  WindDirectionCompass.southWest)
+              .having((w) => w.created, 'created',
+                  DateTime.parse('2020-10-26T00:20:01.840132Z'))
+              .having((w) => w.applicableDate, 'applicableDate',
+                  DateTime.parse('2020-10-26'))
+              .having((w) => w.minTemp, 'minTemp', 7.9399999999999995)
+              .having((w) => w.maxTemp, 'maxTemp', 13.239999999999998)
+              .having((w) => w.theTemp, 'theTemp', 12.825)
+              .having((w) => w.windSpeed, 'windSpeed', 7.876886316914553)
+              .having(
+                  (w) => w.windDirection, 'windDirection', 246.17046093256732)
+              .having((w) => w.airPressure, 'airPressure', 997.0)
+              .having((w) => w.humidity, 'humidity', 73)
+              .having((w) => w.visibility, 'visibility', 11.037727173307882)
+              .having((w) => w.predictability, 'predictability', 73),
+        );
+      });
+    });
+  });
+}
+```
+
+Finally, let's install the `coverage` package to run our tests with coverage and ensure that our tests pass
+
+```yaml
+# packages/meta_weather_api/pubspec.yaml (done)
+name: meta_weather_api
+description: A Dart API Client for the MetaWeather API.
+version: 1.0.0+1
+
+environment:
+  sdk: ">=2.12.0-0 <3.0.0"
+
+dependencies:
+  http: ^0.13.0
+  json_annotation: ^4.0.0
+
+dev_dependencies:
+  coverage: ^0.14.1
+  build_runner: ^1.10.0
+  json_serializable: ^4.0.0
+  mocktail: ^0.1.0
+  test: ^1.16.4
+```
+
+```bash
+flutter test --coverage
+```
+
+## Repository Layer
+
+The goal of our repository layer is to create a wrapper for our data layer, and facilitate communication with the Bloc layer. In doing this, the rest of our code base depends only on functions exposed by our repository layer, instead of specific data provider implementations. This makes it easier for data providers to be switched out in the long run, and even converted into Flutter plugins/packages.
```

Comment:
```suggestion
The goal of our repository layer is to create a wrapper for our data layer and facilitate communication with the Bloc layer. In doing this, the rest of our code base depends only on functions exposed by our repository layer, instead of specific data provider implementations. This makes it easier to switch out data providers later, or even convert them into individual Flutter plugins/packages.
```